File: javamodule/src/week7/section1/implicitConstructorCall/ConstructorCallTest.java
package week7.section1.implicitConstructorCall;

public class ConstructorCallTest {
    public static void main(String[] args) {
        B b1 = new B();
        B b2 = new B(5);
    }
}

class A {
    A() {
        System.out.println("생성자 A : no parameter");
    }

    A(int x) {
        System.out.println("생성자 A  with  parameter x");
    }
}

class B extends A {
    B() {
        System.out.println("생성자 B  with  no parameter");
    }

    B(int x) {
        System.out.println("생성자 B with  parameter x");
    }

}

File: src/main/kotlin/week7/section1/implicitConstructorCall/constructorcalltest.kt
package week7.section1.implicitConstructorCall

open class A {

    constructor(x:Int) {
        println("생성자 A  with  parameter x")
    }
}

class B : A {
    constructor() :super(1){
        println("생성자 B  with  no parameter")
    }
    constructor(x:Int) :super(x){
        println("생성자 B with  parameter x")
    }

}


fun main() {
    val b1 = B()
    val b2 = B(5)


}